utils/s2n_result.c

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/*
+ * The goal of s2n_result is to provide a strongly-typed error
+ * signal value, which provides the compiler with enough information
+ * to catch bugs.
+ *
+ * Historically, s2n has used int to signal errors. This has caused a few issues:
+ *
+ * ## GUARD in a function returning integer types
```

Comment:
integer or unsigned integer?

Reply:
Either one! `-1` is cast to the max value in unsigned integers.